### PR TITLE
Implement verify blockslot forger - Closes #4144

### LINKED ISSUE #4144: Implement verifyBlockForger in Dpos module
### Description

Dpos module should expose a method to validate if a given block was forged by the expected delegate.

### Motivation

As Dpos module is replacing Rounds module, this method is required to replace `round. validateBlockSlot` and `round.validateBlockSlotAgainstPreviousRound`

### Acceptance Criteria

The new method should validate if a block was forged by the expected delegate for any given block that was not finalized yet.

### Additional Information
`feature/bft`

### PR DESCRIPTION
**Pull request by michielmulders**
### What was the problem?
Dpos module should expose a method to validate if a given block was forged by the expected delegate.

### How did I solve it?
Add functionality inside `framework/src/modules/chain/dpos/delegates_list.js` to verify if the block was forged by the right delegate public key.

### How to manually test it?
`npm run jest:unit -- test/jest/unit/specs/modules/chain/dpos/verifyBlockForger.spec.js`

### Review checklist

- [x] The PR resolves #4144 
- [x] All new code is covered with unit tests
- ~[ ] Relevant integration / functional tests are added~
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

### CODE REVIEW

File: framework/test/jest/unit/specs/modules/chain/dpos/verifyBlockForger.spec.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+			await expect(dpos.verifyBlockForger(block)).rejects.toEqual(error);
+		});
+	});
+
+	describe('When DPoS returns empty delegate list', () => {
+		const stubs = {};
+		let dpos;
+		let slots;
+
+		beforeEach(() => {
```
Comment (yatki) [inline framework/test/jest/unit/specs/modules/chain/dpos/verifyBlockForger.spec.js]: actually you don't have to initialise everything from scratch here. You can simply update the return value of the mock in the arrange block of the test ? Example: https://github.com/LiskHQ/lisk-sdk/blob/6ec44cca9ed2d5fed60edcc65058834a350e6aa9/framework/test/jest/unit/specs/modules/chain/dpos/apply.spec.js#L307-L316

Comment (michielmulders) [reply framework/test/jest/unit/specs/modules/chain/dpos/verifyBlockForger.spec.js]: Thanks for pointing that out!

File: framework/src/modules/chain/dpos/delegates_list.js [hunk 2]
```diff
@@ -95,6 +96,35 @@ class DelegatesList extends EventEmitter {
...
+
+		// Get delegate key that forged this block
+		const delegatePubKey = delegateList[currentSlot % this.activeDelegates];
+
+		// Verify if forger exists and matches the generatorPublicKey on block
+		if (delegatePubKey && block.generatorPublicKey === delegatePubKey) {
+			return true;
+		}
+
+		throw new Error(`Failed to verify slot: ${currentSlot}`);
```
Comment (yatki) [inline framework/src/modules/chain/dpos/delegates_list.js]: trivial but I'd refactor the code above as below to honor "always error first" rule.

```js
// Verify if forger exists and matches the generatorPublicKey on block
		if (!delegatePubKey || block.generatorPublicKey !== delegatePubKey) {
			throw new Error(`Failed to verify slot: ${currentSlot}`);
		}

return true;
		
```

File: framework/src/modules/chain/dpos/delegates_list.js [hunk 3]
```diff
@@ -95,6 +96,35 @@ class DelegatesList extends EventEmitter {
...
+	 * @throw {Error} Failed to verify slot
+	 */
+	async verifyBlockForger(block) {
+		const currentSlot = this.slots.getSlotNumber(block.timestamp);
+		const round = this.slots.calcRound(block.height);
+		const delegateList = await this.getRoundDelegates(round);
+
+		if (!delegateList.length) {
+			throw new Error(
+				`Failed to verify slot: ${currentSlot} - No delegateList was found`,
```
Comment (lsilvs) [inline framework/src/modules/chain/dpos/delegates_list.js]:
```suggestion
				`Failed to verify slot: ${currentSlot} for block ID: ${block.id} - No delegateList was found`,
```

File: framework/src/modules/chain/dpos/delegates_list.js [hunk 4]
```diff
@@ -95,6 +96,35 @@ class DelegatesList extends EventEmitter {
...
+			);
+		}
+
+		// Get delegate key that forged this block
```
Comment (lsilvs) [inline framework/src/modules/chain/dpos/delegates_list.js]:
```suggestion
		// Get delegate public key that was supposed to forge the block
```

File: framework/src/modules/chain/dpos/delegates_list.js [hunk 5]
```diff
@@ -95,6 +96,35 @@ class DelegatesList extends EventEmitter {
...
+		const delegatePubKey = delegateList[currentSlot % this.activeDelegates];
```
Comment (lsilvs) [inline framework/src/modules/chain/dpos/delegates_list.js]:
```suggestion
		const expectedForgerPublicKey = delegateList[currentSlot % this.activeDelegates];
```

**Pull request merged**